### DISCUSSION TITLE
[Q/A][Java SDK (Component)]java.util.Collections$UnmodifiableMap lock ,Dubbo thread pool being exhausted

### DISCUSSION THREAD

**Discussion by vanto-** [apache/dubbo#14130] [Question]
### Pre-check

- [X] I am sure that all the content I provide is in English.

### Apache Dubbo Component

Java SDK (apache/dubbo)

### Details

Encountered an issue with the Dubbo thread pool being exhausted. Have you ever encountered this with Dubbo version 3.2.9? Below are the analysis results of the thread status at that time, as well as the details of the [threads.]

'DubboServerHandler-10.200.254.224:20880-thread-9' thread is stuck on line #1750 of java.util.HashMap file in newNode() method. Before getting stuck, this thread obtained 3 locks (java.util.Collections$UnmodifiableMap lock, org.apache.dubbo.config.ReferenceConfig lock...) and never released it. Due to that 176 threads are BLOCKED as shown in this [stack trace](https://fastthread.io/thread.jsp?tId=MTAzOQ==&pc=1&dumpId=1&graph=true&ts=2024-04-25T11-06-31). If threads are BLOCKED for a prolonged period, your application can become unresponsive.

[Dubbo_JStack.txt](https://github.com/apache/dubbo/files/15108722/Dubbo_JStack.txt)

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by AlbumenJ** [comment]
![image](https://github.com/apache/dubbo/assets/9292748/6a3f2a32-fdc0-41e9-8c6f-e0697a5705e3)

Refer service in runtime is not a good idea
